### ISSUE TITLE
[Bug] [API] Modify deleted task will write twice version log

### ISSUE THREAD

**Issue by HomminLee** [apache/dolphinscheduler#14805]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

modify deleted task will write twice version table.

<img width="1230" alt="image" src="https://github.com/apache/dolphinscheduler/assets/25881185/c51ea616-a293-4100-b9b0-e81ad5d3107d">
<img width="641" alt="image" src="https://github.com/apache/dolphinscheduler/assets/25881185/4378eba7-d219-400b-b262-a6ba1a5b0873">

If run this task, will cycle throw sql exception, 10min print 1.5G log.

<img width="1275" alt="image" src="https://github.com/apache/dolphinscheduler/assets/25881185/48e1f8d2-cff2-44ff-8ea0-fef3da708af5">

### What you expected to happen

modify deleted task only write one version table.

### How to reproduce

1. create a workflow contain two task: TaskA, TaskB. You will get workflow version 1.
2. delete TaskA in Task Definition page. You will get workflow version 2.
3. switch workflow to version 1.
4. edit TaskA in workflow DAG page.  You will get workflow version 3 and **two** TaskA version 2.

don't run this workflow, unless you know how to clear through database.

### Version

3.1.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by HomminLee** [comment]
@jieguangzhou Hi, I noticed that there is some extra code in the `dev` branch.
 
```
List<TaskDefinitionLog> newInsertTaskDefinitionLogs = newTaskDefinitionLogs.stream()
                .filter(taskDefinitionLog -> !updateTaskDefinitionLogs.contains(taskDefinitionLog))
                .collect(Collectors.toList());
if (CollectionUtils.isNotEmpty(newInsertTaskDefinitionLogs)) {
    insertResult = taskDefinitionLogMapper.batchInsert(newInsertTaskDefinitionLogs);w
}
```

These code can fix this issue, and it's from pr #13103. This pr can be merge into branch 3.1.x?

**Comment by jieguangzhou** [comment]
Hi @HomminLee Yes, you can submit a PR to fix this
